src/protocols/bdx/tests/TestBdxTransferSession.cpp

```diff
@@ -207,82 +408,237 @@ void TestInitiatingSenderSenderDrive(nlTestSuite * inSuite, void * inContext)
-    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
-    respondingReceiver.PollOutput(outEvent);
-    NL_TEST_ASSERT(inSuite, outEvent.EventType == TransferSession::kOutput_InitReceived);
-    NL_TEST_ASSERT(inSuite, respondingReceiver.GetControlMode() == driveMode);
-    NL_TEST_ASSERT(inSuite, outEvent.transferInitData.FileDesignator != nullptr);
-    NL_TEST_ASSERT(inSuite,
-                   !memcmp(testFileDes, outEvent.transferInitData.FileDesignator, outEvent.transferInitData.FileDesLength));
-
     // Compose SendAccept parameters struct and give to respondingSender
-    char testMetadata[9]       = { "metadata" };
     uint16_t proposedBlockSize = transferBlockSize;
     TransferSession::TransferAcceptData acceptData;
     acceptData.ControlMode    = respondingReceiver.GetControlMode();
     acceptData.MaxBlockSize   = proposedBlockSize;
-    acceptData.StartOffset    = 0;
-    acceptData.Length         = 0;
-    acceptData.Metadata       = reinterpret_cast<uint8_t *>(testMetadata);
-    acceptData.MetadataLength = 9;
+    acceptData.StartOffset    = 0; // not used in SendAccept
+    acceptData.Length         = 0; // not used in SendAccept
+    acceptData.Metadata       = nullptr;
+    acceptData.MetadataLength = 0;
+
+    SendAndVerifyAcceptMsg(inSuite, inContext, outEvent, respondingReceiver, kRole_Receiver, acceptData, initiatingSender,
+                           initOptions);
+
+    // Test multiple Block -> BlockAck -> Block
+    for (int i = 0; i < 3; i++)
+    {
+        SendAndVerifyArbitraryBlock(inSuite, inContext, initiatingSender, respondingReceiver, outEvent, false);
+        SendAndVerifyBlockAck(inSuite, inContext, initiatingSender, respondingReceiver, outEvent, false);
+    }
+
+    SendAndVerifyArbitraryBlock(inSuite, inContext, initiatingSender, respondingReceiver, outEvent, true);
+    SendAndVerifyBlockAck(inSuite, inContext, initiatingSender, respondingReceiver, outEvent, true);
+}
+
+// Test that calls to AcceptTransfer() with bad parameters result in an error.
+void TestBadAcceptMessageFields(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    TransferSession::OutputEvent outEvent;
+    TransferSession initiatingReceiver;
+    TransferSession respondingSender;
+
+    uint16_t maxBlockSize          = 64;
+    TransferControlFlags driveMode = kControl_ReceiverDrive;
+    uint64_t commonLength          = 0;
+    uint64_t commonOffset          = 0;
+    uint32_t timeoutMs             = 1000 * 24;
+
+    // Initialize struct with TransferInit parameters
+    TransferSession::TransferInitData initOptions;
+    initOptions.TransferCtlFlagsRaw = driveMode;
+    initOptions.MaxBlockSize        = maxBlockSize;
+    initOptions.StartOffset         = commonOffset;
+    initOptions.Length              = commonLength;
+    char testFileDes[9]             = { "test.txt" }; // arbitrary file designator
+    initOptions.FileDesLength       = sizeof(testFileDes) - 1;
```

Comment:
Would be clearer here to use `strlen(testFileDes)` for the tests

Reply:
wasn't sure we could use string.h, thanks

Reply:
`strn*` is always OK AFAIK, but `strlen` is fine for a test, especially over a constant string.